jumpscale/packages/tfgrid_solutions/chats/discourse.py

```diff
@@ -0,0 +1,243 @@
+import math
+
+from jumpscale.loader import j
+from jumpscale.sals.chatflows.chatflows import GedisChatBot, chatflow_step, StopChatFlow
+import uuid
+from jumpscale.sals.reservation_chatflow import deployer
+import nacl
+import nacl.signing
+import random
+
+
+class Discourse(GedisChatBot):
+    FLIST_URL = "https://hub.grid.tf/omar0.3bot/omarelawady-discourse_all_in_one-latest.flist"
+
+    steps = [
+        "discourse_start",
+        "discourse_name",
+        "discourse_smtp_server",
+        "discourse_smtp_username",
+        "discourse_smtp_password",
+        "discourse_email",
+        "select_pool",
+        "discourse_network",
+        "overview",
+        "reservation",
+        "discourse_access",
+    ]
+
+    title = "Discourse"
+
+    @chatflow_step()
+    def discourse_start(self):
+        self.solution_id = uuid.uuid4().hex
+        self.user_form_data = dict()
+        self.query = dict()
+        self.user_form_data["chatflow"] = "discourse"
+        self.md_show("# This wizard will help you deploy discourse", md=True)
+        self.threebot_name = j.data.text.removesuffix(self.user_info()["username"], ".3bot")
+        self.solution_metadata = {}
+
+    @chatflow_step(title="Solution name")
+    def discourse_name(self):
+        self.solution_name = deployer.ask_name(self)
+
+    @chatflow_step(title="SMTP server")
+    def discourse_smtp_server(self):
+        self.smtp_server = self.string_ask("Please, enter the smtp server address", required=True)
+
+    @chatflow_step(title="SMTP username")
+    def discourse_smtp_username(self):
+        self.smtp_username = self.string_ask("Please, enter the smtp server username", required=True)
+
+    @chatflow_step(title="SMTP password")
+    def discourse_smtp_password(self):
+        self.smtp_password = self.string_ask("Please, enter the smtp server password", required=True)
+
+    @chatflow_step(title="Email")
+    def discourse_email(self):
+        self.email = deployer.ask_email(self)
+        self.container_resources()
```

Comment:
maybe we can remove this method and just roll its content here, it's only called once

Reply:
It's just a hack to encapsulate work units since only chat flow method gets called. Unrolling the method's content would make a huge method code with unrelated functionality like in method `discourse_network`.

Reply:
as you see (Y)